DMCompiler/DM/Visitors/DMProcBuilder.cs

```diff
@@ -466,6 +466,11 @@ sealed class DMProcBuilder {
                             var outputVar = DMExpression.Create(_dmObject, _proc, outputExpr);
                             var list = DMExpression.Create(_dmObject, _proc, exprIn.List);
 
+                            if (list is {Path: { } path} && path.IsDescendantOf(DreamPath.Area)) {
```

Comment:
Checking the path itself isn't quite correct, since an object's path and its inheritance aren't the same thing. For example:
```
/space
    parent_type = /area
```
The area's path is `/space` but it's still a type of area. You want to instead get the object from `DMObjectTree` and use `IsSubtypeOf()`
```suggestion
                            if (list is {Path: { } path} && DMObjectTree.GetDMObject(path, createIfNonexistent: false)?.IsSubtypeOf(DreamPath.Area) == true) {
```